File: app/src/main/java/com/openmobl/pttDriver/model/DeviceWithStatus.java
package com.openmobl.pttDriver.model;

public class DeviceWithStatus extends Device {
    private static final String TAG = DeviceWithStatus.class.getName();

    private boolean mConnected;
    private int mBatteryLevel;

    public DeviceWithStatus(DeviceType type, String name, String macAddress,
                  boolean autoConnect, boolean autoReconnect, int pttDownDelay) {
        super(type, name, macAddress, autoConnect, autoReconnect, pttDownDelay);

        initialize();
    }

    public DeviceWithStatus(int id, DeviceType type, String name, String macAddress,
                  boolean autoConnect, boolean autoReconnect, int pttDownDelay) {
        super(id, type, name, macAddress, autoConnect, autoReconnect, pttDownDelay);

        initialize();
    }
    public DeviceWithStatus(DeviceType type, String name, String macAddress,
                  int driverId, boolean autoConnect, boolean autoReconnect, int pttDownDelay) {
        super(type, name, macAddress, driverId, autoConnect, autoReconnect, pttDownDelay);

        initialize();
    }
    public DeviceWithStatus(int id, DeviceType type, String name, String macAddress,
                  int driverId, boolean autoConnect, boolean autoReconnect, int pttDownDelay) {
        super(id, type, name, macAddress, driverId, autoConnect, autoReconnect, pttDownDelay);

        initialize();
    }
    public DeviceWithStatus(int id, DeviceType type, String name, String macAddress,
                  int driverId, String driverName, boolean autoConnect, boolean autoReconnect, int pttDownDelay) {
        super(id, type, name, macAddress, driverId, driverName, autoConnect, autoReconnect, pttDownDelay);

        initialize();
    }

    public DeviceWithStatus(Device device) {
        super(device.getId(), device.getDeviceType(), device.getName(), device.getMacAddress(),
                device.getDriverId(), device.getDriverName(),
                device.getAutoConnect(), device.getAutoReconnect(), device.getPttDownDelay());

        initialize();
    }

    private void initialize() {
        mConnected = false;
        mBatteryLevel = 0;
    }

    public void setBatteryLevel(int level) {
        mBatteryLevel = level;
    }
    public int getBatteryLevel() {
        return mBatteryLevel;
    }
    public void setConnected(boolean connected) {
        mConnected = connected;
    }
    public boolean getConnected() {
        return mConnected;
    }
}

File: app/src/main/java/com/openmobl/pttDriver/model/DeviceWithStatus.kt
package com.openmobl.pttDriver.model

import com.openmobl.pttDriver.model.DeviceWithStatus

class DeviceWithStatus : Device {
    var connected = false
    var batteryLevel = 0

    constructor(
        type: DeviceType?, name: String?, macAddress: String?,
        autoConnect: Boolean, autoReconnect: Boolean, pttDownDelay: Int
    ) : super(type, name, macAddress, autoConnect, autoReconnect, pttDownDelay) {
        initialize()
    }

    constructor(
        id: Int, type: DeviceType?, name: String?, macAddress: String?,
        autoConnect: Boolean, autoReconnect: Boolean, pttDownDelay: Int
    ) : super(id, type, name, macAddress, autoConnect, autoReconnect, pttDownDelay) {
        initialize()
    }

    constructor(
        type: DeviceType?, name: String?, macAddress: String?,
        driverId: Int, autoConnect: Boolean, autoReconnect: Boolean, pttDownDelay: Int
    ) : super(type, name, macAddress, driverId, autoConnect, autoReconnect, pttDownDelay) {
        initialize()
    }

    constructor(
        id: Int, type: DeviceType?, name: String?, macAddress: String?,
        driverId: Int, autoConnect: Boolean, autoReconnect: Boolean, pttDownDelay: Int
    ) : super(id, type, name, macAddress, driverId, autoConnect, autoReconnect, pttDownDelay) {
        initialize()
    }

    constructor(
        id: Int,
        type: DeviceType?,
        name: String?,
        macAddress: String?,
        driverId: Int,
        driverName: String?,
        autoConnect: Boolean,
        autoReconnect: Boolean,
        pttDownDelay: Int
    ) : super(
        id,
        type,
        name,
        macAddress,
        driverId,
        driverName,
        autoConnect,
        autoReconnect,
        pttDownDelay
    ) {
        initialize()
    }

    constructor(device: Device) : super(
        device.id, device.deviceType, device.name, device.macAddress,
        device.driverId, device.driverName,
        device.autoConnect, device.autoReconnect, device.pttDownDelay
    ) {
        initialize()
    }

    private fun initialize() {
        connected = false
        batteryLevel = 0
    }

    companion object {
        private val TAG = DeviceWithStatus::class.java.name
    }
}